### PR TITLE
Introduce `call-gen`

### PR DESCRIPTION
**Pull request by SabaunT**
This is needed to divide common logic for `gear-node-fuzzer` and `gear-node-loader`, also it brings simplicity in resolving issues about inter-program communication in the `gear-node-loader`. This changes must be merged before #2183.

This pr basically just moves known entities to another crate.

### CODE REVIEW

**Review by breathx — CHANGES_REQUESTED** [review]

File: utils/call-gen/src/create_program.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+//! Create program call generator.
+
+use crate::{CallGenRng, GearCall, Seed};
+use gear_core::ids::CodeId;
+use gear_utils::{NonEmpty, RingGet};
+
+// code id, salt, payload, gas limit, value
+type CreateProgramArgsInner = (CodeId, Vec<u8>, Vec<u8>, u64, u128);
+
+/// Crate program args
```
Comment (gshep) [inline utils/call-gen/src/create_program.rs:28]: create?

File: utils/call-gen/src/create_program.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+///
+/// Main type used to generate arguments for the `pallet_gear::Pallet::<T>::create_program` call.
+pub struct CreateProgramArgs(pub CreateProgramArgsInner);
+
+impl From<CreateProgramArgs> for CreateProgramArgsInner {
+    fn from(
+        CreateProgramArgs((code_id, salt, payload, gas_limit, value)): CreateProgramArgs,
```
Comment (gshep) [inline utils/call-gen/src/create_program.rs:35]: destructuring in functions signature looks too noisy. I would suggest to use `let ...` inside the body

File: utils/call-gen/src/create_program.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    fn from(args: CreateProgramArgs) -> Self {
+        GearCall::CreateProgram(args)
+    }
+}
+
+impl TryFrom<GearCall> for CreateProgramArgs {
+    type Error = ();
+
+    fn try_from(call: GearCall) -> Result<Self, Self::Error> {
+        if let GearCall::CreateProgram(call) = call {
```
Comment (gshep) [inline utils/call-gen/src/create_program.rs:51]: `match` should be shorter imho

**Review by gshep — APPROVED** [review]
Good job! 👍

File: utils/call-gen/src/create_program.rs (hunk 3 above)
Comment (SabaunT) [reply utils/call-gen/src/create_program.rs:51]: discussed.

File: utils/call-gen/src/create_program.rs (hunk 2 above)
Comment (SabaunT) [reply utils/call-gen/src/create_program.rs]: done

File: utils/call-gen/src/create_program.rs (hunk 1 above)
Comment (SabaunT) [reply utils/call-gen/src/create_program.rs]: done

File: utils/utils/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+[package]
+name = "gear-utils"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+nonempty = "0.8.1"
```
Comment (breathx) [inline utils/utils/Cargo.toml:7]: newline

Comment (SabaunT) [reply utils/utils/Cargo.toml]: done

**Pull request merged**